lms/services/lti_names_roles.py

```diff
@@ -36,14 +33,23 @@ def __init__(self, ltia_http_service: LTIAHTTPService):
         self._ltia_service = ltia_http_service
 
     def get_context_memberships(
-        self, lti_registration: LTIRegistration, service_url: str
+        self,
+        lti_registration: LTIRegistration,
+        service_url: str,
+        resource_link_id: str | None = None,
     ) -> list[Member]:
         """
-        Get all the memberships of a context (a course).
+        Get the roster for a course or assignment.
 
         The course is defined by the service URL which will obtain
         from a LTI launch parameter and is always linked to an specific context.
+
+        Optically, using the  same service_url the API allows to get the roster of an assignment identified by `resource_link_id`.
         """
+        query = {}
+        if resource_link_id:
+            query["rlid"] = resource_link_id
```

Comment:
rlid is the name in the API but that's a bit cryptic. It's called resource link ID everywhere else.

Reply:
Perhaps add a comment there to clarify that for the next person? Even adding a link to some documentation would be good, if such thing exists.

Reply:
Yes, added a note to the issue for this, I'll do a follow up to avoid some rebase hell.